### ISSUE TITLE
Add key prop to options

### ISSUE THREAD

**Issue by asbjornenge** [asbjornenge/react-datalist#8]

**Comment by aarkerio** [comment]
Its is possible to add keys to get:

    <option value="123">Sam</option>  
     <option value="234">Tom</option>

**Comment by asbjornenge** [comment]
@aarkerio yeah, that should be doable and is probably what this issue was created for. I haven't worked on this for ages and I don't have the time to dig into it right now. But feel free to submit a [pull request](https://help.github.com/articles/using-pull-requests/) for this and I'll be sure to validate and merge :smile:

**Comment by iswanj** [comment]
@asbjornenge Can you add `hacktoberfest` label for all issues
